mono/mini/interp/transform.c

```diff
@@ -315,8 +421,9 @@ shift_op(TransformData *td, int mint_op)
 }
 
 static int 
-mint_type(MonoType *type)
+mint_type(MonoType *type_)
 {
+	MonoType *type = mini_native_type_replace_type (type_);
```

Comment:
instead of replacing it in the callee, wouldn't it be better to replace it in the caller?